include/libserver/registry/HorseRegistry.hpp

```diff
@@ -74,22 +123,91 @@ class HorseRegistry
   void GiveHorseRandomPotential(
     data::Horse::Potential& potential);
 
+  //! Gets coat information for a given coat TID.
+  //! @param coatTid Coat template ID.
+  //! @returns Reference to Coat, with fallback to default if not found.
+  const Coat& GetCoatInfo(data::Tid coatTid) const;
+
+  //! Gets a random mane TID from the specified color group and shape.
+  //! @param colorGroupId Color group ID.
+  //! @param shape Mane shape (0-7).
+  //! @param rng Random number generator.
+  //! @returns Mane TID, or InvalidTid if not found.
+  data::Tid GetRandomManeFromColorAndShape(int32_t colorGroupId, int32_t shape, std::mt19937& rng) const;
```

Comment:
the random engine should be a member of this class, not passed as a function parameter